cppcache/src/util/make_unique.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef UTIL_MAKE_UNIQUE_H
+#define UTIL_MAKE_UNIQUE_H
+
+#include <cstddef>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+// N3656 standard revision of make_unique.
+
```

Comment:
Should we put this in a namespace?

Reply:
Have a look at how it was done in this other pending PR:  https://github.com/apache/geode-native/pull/891/files

Might be good to ping @gaussianrecurrence and coordinate before there's a conflict, or two definitions in the code base(!).

Reply:
There is an internal shared library, that used go be static linked into the main library, but is still linked into all the test libraries. It was intended to be a place to put common utils that both the main lib and test code could use without exporting them as symbols from the main lib. It is the perfect place to consolidate `make_unique` for both the main lib and tests. The two files currently in this library are obsolete and should be deleted. It would also need to be added back as a library dependency to `_apache-geode` library.

Reply:
When I say "internal shared library" I don't me it is compile as a shared library. It is compiled static and linked static.

Reply:
I'm going to update the PR to address the rest and work on this one tomorrow.